CONTRIBUTING.md

```diff
@@ -73,17 +73,61 @@ Path:       /home/linuxbrew/.linuxbrew/opt/python@3.9
 ```
 If you want to extract only the path to your new virtualenv, you can run `poetry env info --path`
 
-## Loading your virtualenv in another shell 
+### Step 4. Enable pre-commit hooks.
 
-It's a common need when developing against this client to load it as part of some other application or Jupyter Notebook code, modify 
-it directly, see your changes reflected immediately and also have your changes tracked in git so you can contribute them back. 
+Run `poetry run pre-commit install` to enable checks to run when you commit so you don't have to find out during your CI run that minor lint issues need to be addressed.
 
-It's important to understand that, by default, if you open a new shell or terminal window, or, for example, a new pane in a tmux session, 
-your new shell will not yet reference the new virtualenv you created in the previous step. 
+## Common tasks
+
+### Running tests
+
+- Unit tests: `make test-unit`
+- Integration tests: `PINECONE_API_KEY="YOUR API KEY" make test-integration`
+- Run the tests in a single file: `poetry run pytest tests/unit/data/test_bulk_import.py -s -vv`
+
+### Running the ruff linter / formatter
+
+These should automatically trigger if you have enabled pre-commit hooks with `poetry run pre-commit install`. But in case you want to trigger these yourself, you can run them like this:
+
+```
+poetry run ruff check --fix # lint rules
+poetry run ruff format      # formatting
+```
+
+If you want to adjust the behavior of ruff, configurations are in `pyproject.toml`.
```

Comment:
1000% nit, but you probs want to put `ruff` in backticks, or else it's sorta confusing to read at first glance